src/inmanta/graphql/schema.py

```diff
@@ -12,27 +12,48 @@
 Contact: code@inmanta.com
 """
 
+import dataclasses
 import typing
 
 import inmanta.data.sqlalchemy as models
 import strawberry
 from inmanta.data import get_session, get_session_factory
+from inmanta.server.services.compilerservice import CompilerService
 from sqlalchemy import Select, asc, desc, select
 from strawberry import relay
 from strawberry.schema.config import StrawberryConfig
 from strawberry.types import Info
 from strawberry.types.info import ContextType
 from strawberry_sqlalchemy_mapper import StrawberrySQLAlchemyLoader, StrawberrySQLAlchemyMapper
 
+
+@dataclasses.dataclass
+class GraphQLMetadata:
+    compiler_service: CompilerService
+
+
 mapper: StrawberrySQLAlchemyMapper[typing.Any] = StrawberrySQLAlchemyMapper()
 SCHEMA: strawberry.Schema | None = None
+METADATA: GraphQLMetadata | None = None
```

Comment:
I would get rid of these globals and wrap this whole module in a method, ... 

less change of leaks

Reply:
Is this what you meant? 
 - I couldn't extract `Environment` because I need the metadata for the resolver field (I don't think I can pass it as an argument, or at least I haven't figured out how to do it)
 - We are not caching the Schema, we are building it every time we call `get_schema`. Don't know if it impacts performance